pkg/workloadmeta/collectors/ecs/ecs.go

```diff
@@ -221,3 +196,51 @@ func (c *collector) parseTaskContainers(task v1.Task) ([]workloadmeta.Orchestrat
 
 	return taskContainers, events
 }
+
+// getResourceTags fetches task and container instance tags from the ECS API,
+// and caches them for the lifetime of the task, to avoid hitting throttling
+// limits from tasks being updated on every pull. Tags won't change in the
+// store even if they're changed in the resources themselves, but at least that
+// matches the old behavior present in the tagger.
+func (c *collector) getResourceTags(ctx context.Context, entity *workloadmeta.ECSTask) resourceTags {
+	rt, ok := c.resourceTags[entity.ID]
```

Comment:
are we sure that the ecs task info are immutable? 
`task_version` `family` for sure. but do we extra other tag from the ecs task, that can be set dynamically by the customer?

Reply:
Nothing is immutable, that's why we always parse the main task data now. `getResourceTags` isn't immutable either, but caching it matches the behavior we already had in the tagger before workloadmeta (so the same behavior as 7.32). Re-fetching it on every pull will hit throttling limits (that was the original behavior before we fixed the throttling issues), and I think having an expiry mechanism would be out of scope for a bugfix release (I can work on it for 7.35 though).

Reply:
ok that make sens.